### PR TITLE
Prometheus endpoint config for hawkular-metrics and cassandra

### PR DESCRIPTION
**Pull request by jsanda**
Prometheus endpoints are exposed in hawkular-metrics and in cassandra by default. This PR adds options to disable them.

### CODE REVIEW

File: roles/openshift_metrics/defaults/main.yaml [hunk 1]
```diff
@@ -59,3 +59,6 @@ openshift_metrics_cassandra_pvc_access: "{{ openshift_hosted_metrics_storage_acc
 openshift_metrics_hawkular_user_write_access: False
 
 openshift_metrics_heapster_allowed_users: system:master-proxy
+
+openshift_metrics_cassandra_disable_prometheus_endpoint: ""
+openshift_metrics_hawkular_disable_prometheus_endpoint: ""
```
Comment (mwringe) [inline roles/openshift_metrics/defaults/main.yaml]: Should these not be set to false by default?

Comment (jsanda) [inline roles/openshift_metrics/defaults/main.yaml]: No because I just check a zero-length string.

Comment (jsanda) [inline roles/openshift_metrics/defaults/main.yaml]: After further discussion I will change these properties to be false by default.

**Comment by mwringe** [comment]
This looks good to me.

@jcantrill @ewolinetz can one of you please review this?

File: roles/openshift_metrics/defaults/main.yaml [hunk 2]
```diff
@@ -59,3 +59,6 @@ openshift_metrics_cassandra_pvc_access: "{{ openshift_hosted_metrics_storage_acc
 openshift_metrics_hawkular_user_write_access: False
 
 openshift_metrics_heapster_allowed_users: system:master-proxy
+
+openshift_metrics_cassandra_disable_prometheus_endpoint: False
```
Comment (jcantrill) [inline roles/openshift_metrics/defaults/main.yaml]: Wouldn't it make more sense to invert the language so its easier to understand and read, no double negatives: 'openshift_metrics_cassandra_enable_prometheus_endpoint: True'

Comment (jsanda) [inline roles/openshift_metrics/defaults/main.yaml]: I agree that double negatives are confusing. I will change this.

File: roles/openshift_metrics/defaults/main.yaml [hunk 3]
```diff
@@ -59,3 +59,6 @@ openshift_metrics_cassandra_pvc_access: "{{ openshift_hosted_metrics_storage_acc
 openshift_metrics_hawkular_user_write_access: False
 
 openshift_metrics_heapster_allowed_users: system:master-proxy
+
+openshift_metrics_cassandra_enable_prometheus_endpoint: True
```
Comment (ewolinetz) [inline roles/openshift_metrics/defaults/main.yaml]: is there ever a time where we one of these would be true and the other would be false?

Comment (mwringe) [inline roles/openshift_metrics/defaults/main.yaml]: Someone could want to collect metrics just for one or not the other. Or we may want to disable metrics in one to see if that is causing problems.

Comment (ewolinetz) [inline roles/openshift_metrics/defaults/main.yaml]: @mwringe ack, would that be a scenario at installation or post installation? I'm just trying to figure out if this should be a single variable or two. If they are mutually exclusive then it makes sense to keep them as two.

Comment (mwringe) [reply roles/openshift_metrics/defaults/main.yaml]: @ewolinetz sorry, I lost track of this pr, these are separate independent variables, we should have 2.

**Comment by mwringe** [comment]
@ewolinetz is there anything else we need to get done for this issue? I would like to get this closed soon.

**Comment by ewolinetz** [comment]
@mwringe i'll put this through the testing paces today

**Comment by mwringe** [comment]
@ewolinetz anything else we need to do to get this merged?

**Comment by ewolinetz** [comment]
@mwringe with that latest flag the bot should come back around and pick it up and then drop it into the merge queue

**Comment by sdodson** [comment]
Merge gating on kind has been enabled now so from devcut through the end of the sprint we're only supposed to be merging bugfixes.

**Pull request merged**